src/jit/emit.cpp

```diff
@@ -5533,6 +5562,111 @@ void emitter::emitOutputDataSec(dataSecDsc* sec, BYTE* dst)
     }
 }
 
+#ifdef DEBUG
+void emitter::emitDispDataSec(dataSecDsc* section)
+{
+    printf("\n");
+
+    unsigned offset = 0;
+
+    for (dataSection* data = section->dsdList; data != nullptr; data = data->dsNext)
+    {
+        const char* labelFormat = "%-7s";
+        char        label[64];
+        sprintf_s(label, _countof(label), "RWD%02u", offset);
+        printf(labelFormat, label);
+        offset += data->dsSize;
+
+        if ((data->dsType == dataSection::blockRelative32) || (data->dsType == dataSection::blockAbsoluteAddr))
+        {
+            insGroup* igFirst    = static_cast<insGroup*>(emitCodeGetCookie(emitComp->fgFirstBB));
+            bool      isRelative = (data->dsType == dataSection::blockRelative32);
+            size_t    blockCount = data->dsSize / (isRelative ? 4 : TARGET_POINTER_SIZE);
+
+            for (unsigned i = 0; i < blockCount; i++)
+            {
+                if (i > 0)
+                {
+                    printf(labelFormat, "");
+                }
+
+                BasicBlock* block = reinterpret_cast<BasicBlock**>(data->dsCont)[i];
+                insGroup*   ig    = static_cast<insGroup*>(emitCodeGetCookie(block));
+
+                const char* blockLabelFormat = "G_M%03u_IG%02u";
+                char        blockLabel[64];
+                char        firstLabel[64];
+                sprintf_s(blockLabel, _countof(blockLabel), blockLabelFormat, Compiler::s_compMethodsCount, ig->igNum);
+                sprintf_s(firstLabel, _countof(firstLabel), blockLabelFormat, Compiler::s_compMethodsCount,
+                          igFirst->igNum);
+
+                if (isRelative)
+                {
+                    if (emitComp->opts.disDiffable)
+                    {
+                        printf("dd\t%s - %s\n", blockLabel, firstLabel);
+                    }
+                    else
+                    {
+                        printf("dd\t%08Xh", ig->igOffs - igFirst->igOffs);
```

Comment:
Even in non-diffable format would it be useful to print the labels as well?

Reply:
Hah, I forgot that it does something like this already. Here's the diffable output:
```
RWD00  dd       G_M55886_IG03 - G_M55886_IG02
       dd       G_M55886_IG05 - G_M55886_IG02
       dd       G_M55886_IG07 - G_M55886_IG02
       dd       G_M55886_IG09 - G_M55886_IG02
       dd       G_M55886_IG10 - G_M55886_IG02
```
and the non-diffable output:
```
RWD00  dd       00000023h ; case G_M55886_IG03
       dd       00000029h ; case G_M55886_IG05
       dd       0000002Fh ; case G_M55886_IG07
       dd       00000035h ; case G_M55886_IG09
       dd       0000003Ch ; case G_M55886_IG10
```